{
  "affected_release": null,
  "package_state": [
    {
      "product_name": "Red Hat Enterprise Linux 6",
      "fix_state": "Not affected",
      "package_name": "kernel",
      "cpe": "cpe:/o:redhat:enterprise_linux:6"
    },
    {
      "product_name": "Red Hat Enterprise Linux 7",
      "fix_state": "Not affected",
      "package_name": "kernel",
      "cpe": "cpe:/o:redhat:enterprise_linux:7"
    },
    {
      "product_name": "Red Hat Enterprise Linux 7",
      "fix_state": "Not affected",
      "package_name": "kernel-rt",
      "cpe": "cpe:/o:redhat:enterprise_linux:7"
    },
    {
      "product_name": "Red Hat Enterprise Linux 8",
      "fix_state": "Not affected",
      "package_name": "kernel",
      "cpe": "cpe:/o:redhat:enterprise_linux:8"
    },
    {
      "product_name": "Red Hat Enterprise Linux 8",
      "fix_state": "Not affected",
      "package_name": "kernel-rt",
      "cpe": "cpe:/o:redhat:enterprise_linux:8"
    },
    {
      "product_name": "Red Hat Enterprise Linux 9",
      "fix_state": "Not affected",
      "package_name": "kernel",
      "cpe": "cpe:/o:redhat:enterprise_linux:9"
    },
    {
      "product_name": "Red Hat Enterprise Linux 9",
      "fix_state": "Not affected",
      "package_name": "kernel-rt",
      "cpe": "cpe:/o:redhat:enterprise_linux:9"
    }
  ],
  "threat_severity": "Moderate",
  "public_date": "2025-05-01T00:00:00Z",
  "bugzilla": {
    "description": "kernel: RDMA/hns: Fix NULL pointer problem in free_mr_init()",
    "id": "2363402",
    "url": "https://bugzilla.redhat.com/show_bug.cgi?id=2363402"
  },
  "cvss": {
    "cvss_base_score": "",
    "cvss_scoring_vector": "",
    "status": ""
  },
  "cvss3": {
    "cvss3_base_score": "5.5",
    "cvss3_scoring_vector": "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:N/A:H",
    "status": "draft"
  },
  "iava": "",
  "cwe": "",
  "statement": "",
  "acknowledgement": "",
  "name": "CVE-2022-49930",
  "document_distribution": "",
  "details": [
    "In the Linux kernel, the following vulnerability has been resolved:\nRDMA/hns: Fix NULL pointer problem in free_mr_init()\nLock grab occurs in a concurrent scenario, resulting in stepping on a NULL\npointer.  It should be init mutex_init() first before use the lock.\nUnable to handle kernel NULL pointer dereference at virtual address 0000000000000000\nCall trace:\n__mutex_lock.constprop.0+0xd0/0x5c0\n__mutex_lock_slowpath+0x1c/0x2c\nmutex_lock+0x44/0x50\nfree_mr_send_cmd_to_hw+0x7c/0x1c0 [hns_roce_hw_v2]\nhns_roce_v2_dereg_mr+0x30/0x40 [hns_roce_hw_v2]\nhns_roce_dereg_mr+0x4c/0x130 [hns_roce_hw_v2]\nib_dereg_mr_user+0x54/0x124\nuverbs_free_mr+0x24/0x30\ndestroy_hw_idr_uobject+0x38/0x74\nuverbs_destroy_uobject+0x48/0x1c4\nuobj_destroy+0x74/0xcc\nib_uverbs_cmd_verbs+0x368/0xbb0\nib_uverbs_ioctl+0xec/0x1a4\n__arm64_sys_ioctl+0xb4/0x100\ninvoke_syscall+0x50/0x120\nel0_svc_common.constprop.0+0x58/0x190\ndo_el0_svc+0x30/0x90\nel0_svc+0x2c/0xb4\nel0t_64_sync_handler+0x1a4/0x1b0\nel0t_64_sync+0x19c/0x1a0"
  ],
  "references": [
    "https://www.cve.org/CVERecord?id=CVE-2022-49930\nhttps://nvd.nist.gov/vuln/detail/CVE-2022-49930\nhttps://lore.kernel.org/linux-cve-announce/2025050112-CVE-2022-49930-91ab@gregkh/T"
  ]
}